pkg/cluster/cluster.go

```diff
@@ -295,6 +295,8 @@ func (c *Cluster) run(stopC <-chan struct{}) {
 				break
 			}
 
+			garbagecollection.CollectDeadMemberServices(c.cluster.Name, c.cluster.Namespace, c.config.KubeCli, c.members, c.logger)
```

Comment:
See https://github.com/coreos/etcd-operator/pull/680#discussion_r98061829

> We need to sync it with membership.

We can't have membership changed concurrently. The func only preserves the services found in current membership.